storage/src/main/java/org/apache/kafka/storage/internals/log/LocalLog.java

```diff
@@ -470,8 +470,8 @@ public FetchDataInfo read(long startOffset,
         return maybeHandleIOException(
                 () -> "Exception while reading from " + topicPartition + " in dir " + dir.getParent(),
                 () -> {
-                    logger.trace("Reading maximum $maxLength bytes at offset {} from log with total length {} bytes",
-                            startOffset, segments.sizeInBytes());
+                    logger.trace("Reading maximum {} bytes at offset {} from log with total length {} bytes",
+                            maxLength, startOffset, segments.sizeInBytes());
```

Comment:
nit: Since `segments.sizeInBytes()` iterates through all segments to calculate the size, consider adding an `isTraceEnabled` flag to avoid unnecessary loops when trace logging is disabled by default.

@gongxuanzhang WDYT?

Reply:
This is a good catch - looks like we regressed here. The Scala logging library handled this automatically.

Reply:
+1! Nice catch @chia7712 !

Reply:
#19386